geo/src/main/java/org/odk/collect/geo/selection/SelectionMapFragment.kt

```diff
@@ -265,20 +266,25 @@ class SelectionMapFragment(
     }
 
     private fun onFeatureClicked(featureId: Int, maintainZoom: Boolean = true) {
-        val selectedFeatureId = selectedFeatureViewModel.getSelectedFeatureId()
-        if (selectedFeatureId != null && selectedFeatureId != featureId) {
-            resetIcon(selectedFeatureId)
-        }
-
         val item = itemsByFeatureId[featureId]
+        val selectedItem = selectedItemViewModel.getSelectedItem()
+
         if (item != null) {
-            if (!skipSummary) {
-                val point = item.points[0]
+            if (selectedItem != null && selectedItem.id != item.id) {
+                resetIcon(selectedItem)
+            }
 
-                if (maintainZoom) {
-                    map.zoomToPoint(MapPoint(point.latitude, point.longitude), map.zoom, true)
+            if (!skipSummary) {
+                if (item.points.size > 1) {
+                    map.zoomToBoundingBox(item.points, 0.8, true)
```

Comment:
We're doing a "zoom to fit" when selecting a shape/trace rather than maintaining the user zoom. I think that addresses some of the feedback around it being hard to see which item is selected (without having to deal with color/shape changes), but as @lognaturel has pointed out to me, it isn't great in situations where the user taps on the wrong item and then loses context.